### DISCUSSION TITLE
Different BPM/Time Signature for live set tracks

### DISCUSSION THREAD

**Discussion by f00br** [helgoboss/helgobox#1379] [Q&A] [answered]
Hi! I wanted to find out for sure what is the right and correct method to use Playtime 2 in this situation:
I need to use Playtime2 for a live set. I use several VSTi and linear instruments, and I need Playtime 2 to play pre-recorded wav files. 
The thing is that each of these tracks I play has a different BPM and in some cases a different time signature.

Can you give recommendations on how I should set up the project correctly in such a case? Do I have to create a separate Playtime 2 instance for each track and specify for each track the BPM/Time Signature I need or can I somehow organize everything in one Playtime2 instance?

Thank you kindly❤

**Comment by helgoboss** [comment, marked as answer]
Hi!

There's currently no really good way to achieve what you are after. As soon as #1024 is implemented, that would probably be the way.

Until then, here are a few ideas.

## Change tempo and trigger scene on button press

1. Use one mapping per track. Source: Some button. Target: "Project: Set tempo". Glue: Target min/max set to desired tempo.
2. Create more mappings to let the same buttons also trigger target "Playtime: Row action" with action "Play".

## Use REAPER timeline

Here's a solution that doesn't require Playtime at all.

1. Add tempo markers and normal markers to the project and put the audio files at the respective positions.
2. Use ReaLearn target "Marker/region: Go to" to jump to the specific markers

PS: Separate Playtime instances in one project all share the same tempo and time signature, so this is not a solution.

**Reply by f00br** [reply]
Got it. Thanks for the explanation.
But could you describe the first method in a little more detail? I'm not quite sure I understand you correctly.

That is, I arrange the stems of my tracks in playtime and say assign to the button that triggers, for example, drums, tempo change. It turns out that every time I start slots with drums it also changes the tempo?

**Reply by helgoboss** [reply]
The first solution involves a hardware controller with buttons. Do you have that?

**Reply by f00br** [reply]
Yes, I use AKAI APC Mini

**Reply by helgoboss** [reply]
Okay, which buttons would you like to use to switch between the song parts?

**Reply by f00br** [reply]
<img width="566" alt="Hn1Noh8uLH" src="https://github.com/user-attachments/assets/a2731161-4f1e-46be-8f29-d04431822a9b" />

Classically, track stems are switched using the grid buttons

**Reply by f00br** [reply]
Or do you mean the buttons for switching scenes? I just don't use those buttons, i.e. abruptly don't switch scenes only parts (stems)

**Reply by helgoboss** [reply]
Ah, you make a difference between parts/stems and scenes? What is the difference?

Maybe the best thing is if you show me some video what you try to do. If it's a common thing, there should be a YouTube video somewhere showing this process (with another software).

**Reply by f00br** [reply]
<img width="354" alt="9JVrkxkG4k" src="https://github.com/user-attachments/assets/f4fdf5b7-9943-4e8e-9b80-57a126bdf7ee" />

No big deal, sorry to confuse you. What I meant was that I had never used the scene switches (circled in red in the screenshot) until now. I switched between parts of the track using the slots in the main columns.

But now I'm willing to change my settings any way I can to use playtime in my performances.

**Comment by f00br** [comment]
Okay, if I understand the principle correctly. 
I create a series of scripts that change the tempo of the project
(reaper.SetCurrentBPM(0, 120.0, true))
I set Timebase for item: Time

I bind the script to the controller button that starts the very first stem in my track.

What do you think of this plan?

**Reply by f00br** [reply]
![image](https://github.com/user-attachments/assets/6ac08b3f-c786-439c-be07-7ec86d10db39)

And it turns out that when the option (on the screenshot) is enabled the controller stops responding to the fact that I have configured scripts for it in Actions

**Reply by helgoboss** [reply]
> Okay, if I understand the principle correctly. 
> I create a series of scripts that change the tempo of the project
> (reaper.SetCurrentBPM(0, 120.0, true))
> I set Timebase for item: Time
> 
> I bind the script to the controller button that starts the very first stem in my track.
> 
> What do you think of this plan?

That would work. But if you just wanted to change tempo (not time signature), you could also use target "Project: Set tempo". It's easier.

**Reply by helgoboss** [reply]
> ![image](https://github.com/user-attachments/assets/6ac08b3f-c786-439c-be07-7ec86d10db39)
> 
> And it turns out that when the option (on the screenshot) is enabled the controller stops responding to the fact that I have configured scripts for it in Actions

If this option is enabled, Playtime will load the default preset for your APC.

The big question is: Do you want to use the default preset plus your own mappings or do you want just your own mappings?

Using ReaLearn is a science on its own. I'm not sure how you have configured it.

**Reply by f00br** [reply]
> > ![image](https://private-user-images.githubusercontent.com/43912817/398931666-6ac08b3f-c786-439c-be07-7ec86d10db39.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzUzMDI3OTMsIm5iZiI6MTczNTMwMjQ5MywicGF0aCI6Ii80MzkxMjgxNy8zOTg5MzE2NjYtNmFjMDhiM2YtYzc4Ni00MzljLWJlMDctN2VjODZkMTBkYjM5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjI3VDEyMjgxM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTljZDIwMDRkN2UzZjZkNDk2ZDBmMTcyMmJiZjI0YzI2ZjNiZmQxYjlmZDMxMDMzZGNkNDAxNzVjOWNhZTk1MjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.tPHHofaDvmTCHvhpR3xKqh7NQbaGtLYkN9IWZqStaTs)
> > And it turns out that when the option (on the screenshot) is enabled the controller stops responding to the fact that I have configured scripts for it in Actions
> 
> If this option is enabled, Playtime will load the default preset for your APC.
> 
> The big question is: Do you want to use the default preset plus your own mappings or do you want just your own mappings?
> 
> Using ReaLearn is a science on its own. I'm not sure how you have configured it.

Well right now I'd like to use the default preset plus my own mappings (somehow), since I don't know ReaLearn at all. It would be great to use the default preset for the controller and be able to bind tempo changes to some individual buttons. But if there is no such possibility, I can find free buttons on other controllers (I also use a foot midi controller).

**Reply by helgoboss** [reply]
In order to create a custom controller experience involving Playtime and other advanced targets such as "Project: Invoke REAPER action" or "Project: Set tempo", you'll need to get familiar with ReaLearn. If you don't know it yet, I strongly suggest to take the easier route first: Map the tempo changes to another controller.

For this, you can use the same Helgobox instance as the one that hosts Playtime. And within that instance, create your ReaLearn mappings in the main unit.

To build these mappings and to understand the general concept of ReaLearn, please watch the ReaLearn tutorial videos.
